packages/astro-tests/src/astroFixture.ts

```diff
@@ -90,17 +91,35 @@ type Fixture = {
 	 */
 	pathExists: (path: string) => boolean;
 	/**
-	 * Read a file from the build.
+	 * Read a unicode text file from the build. Do NOT use this for non-text files (e.g. images).
 	 *
 	 * Returns null if the file doesn't exist.
 	 */
 	readFile: (path: string) => Promise<string | null>;
 	/**
-	 * Read a file from the project.
+	 * Read a file from the build. This is a thin wrapper around `fs.promises.readFile` and will not modify the encoding, making it suitable for reading binary files (e.g. images).
+	 *
+	 * Returns null if the file doesn't exist.
+	 */
+	readRawFile: (
+		path: string,
+		options?: Parameters<typeof readFile>[1]
+	) => Promise<null | Awaited<ReturnType<typeof readFile>>>;
```

Comment:
These types are not too convenient. `readFile` allows passing the encoding on the options, and that guarantees a `string` output. Not passing the encoding guarantee a `Buffer` output.

This type will always return `string | Buffer`.

I think it would be better to add an optional encoding parameter to `readFile` and `readSrcFile`, with `utf-8` as default to match the current behavior, such that it always return a string.

Then these raw functions can receive just the path and always return a Buffer.

Reply:
Happy to make any changes you want but not sure I understand what result you’d like.

>  I think it would be better to add an optional encoding parameter to readFile and readSrcFile, with utf-8 as default to match the current behavior, such that it always return a string.

If I add an encoding parameter, even if it defaults to utf8, the method will no longer only return string?

Reply:
Oh, did you mean an optional parameter, but one that is always a `string`? Something like this?:

```ts
readFile(path: string, encoding?: string = "utf8"): Promise<string> { ...  return readFile(path, encoding) },
readFileAsBuffer(path: string): Promise<Buffer> { ... return readFile(path) },
```

Ok yeah I think that's what you meant and makes sense to me. I can whip that up and push an update.

Reply:
Okay, re-requesting review. I *think* I made the change you wanted and I added the changeset.